selector_table.cc

```diff
@@ -294,20 +263,11 @@ struct SelectorHash
 			hash = hash * 33 + c;
 		}
 #ifdef TYPE_DEPENDENT_DISPATCH
-		// We can't use all of the values in the type encoding for the hash,
-		// because our equality test is a bit more complex than simple string
-		// encoding (for example, * and ^C have to be considered equivalent, since
-		// they are both used as encodings for C strings in different situations)
```

Comment:
Are you sure clang never does this?  I vaguely remember things like `char*` and `[un]signed char*` encoded differently.